### DISCUSSION TITLE
server side JS ORM example?

### DISCUSSION THREAD

**Discussion by dcsan** [supabase/supabase#3726] [Questions]
it seems the supabase JS lib provides it's own lightweight chaining syntax, a bit like an ORM.

are there any examples of using more mainstream JS ORMs, or prisma / typeORM ?

eg using row-level authentication etc seems like it would need some setup.

**Comment by silentworks** [comment]
If you are using a ORM like prisma or TypeORM you would be using the database connection directly hence bypassing RLS. Also this would kinda remove some of the benefits of Supabase as a platform. Think of the platform as being aimed at front end developers who want a database to work with, even though it can be used by backend devs too since you have direct access to the database itself.

**Reply by fmatuszewski** [reply]
Exactly. But there are some open source projects that use TypeORM that I would like to integrate with supabase. I want to create a middleware that passes jwt with each database query .
I want to integrate Medusa with supabase.
To integrate this I need replace 
passport - with supabase client. 
The best option would be to create passport supabase strategy to provide easy transition from projects that use passport to supabase.

The second thing is to provide middle ware that sets up required postgresql variables with every request.
from what i found in https://github.com/supabase/auth/blob/master/migrations/20220531120530_add_auth_jwt_function.up.sql
this variables are set by supabase:
nullif(current_setting('request.jwt.claim', true), ''),
        nullif(current_setting('request.jwt.claims', true), '')
    )::jsonb
but I could not find the exact piece of code that sets them.
Can you assist me with that.
1) consider creating midlleware library that will help to integrate passport typeorm with supabase. or
2) help me find a code resposible for setting request.jwt.claims or whole request in postgresql session. or
3) provide me information what is json structure of request variable and or request.jwt.claims variable
